src/App.tsx

```diff
@@ -1,60 +1,106 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable react/jsx-no-comment-textnodes */
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  function getNewPostId(todos: Todo[]) {
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    return maxId + 1;
+  }
+
+  const addNewTodo = (newTodo: Todo) => {
+    setTodos(todos => [...todos, newTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (evnt: React.FormEvent) => {
+    evnt.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(userId === 0);
+
+    if (!title.trim() || !userId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNewPostId(todos),
+      title,
+      userId,
+      completed: false,
+    };
+
+    addNewTodo(newTodo);
+
+    reset();
+  };
+
+  const handleUserChange = (evnt: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+evnt.target.value);
+
+    setHasUserError(+evnt.target.value === 0);
+  };
+
+  const handleTitleChange = (evnt: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(evnt.target.value);
+    setHasTitleError(!title.trim());
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
+      // #region Form
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            value={title}
+            data-cy="titleInput"
+            onChange={handleTitleChange}
+          />
+          {hasTitleError && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
+          <select
+            data-cy="userSelect"
+            value={userId}
+            onChange={handleUserChange}
+          >
+            <option value="0">Choose a user</option>
+            {usersFromServer.map(user => (
+              <option key={user.id} value={user.id}>
+                {user.name}
+              </option>
+            ))}
```

Comment:
[CODE STYLE] - Prepare data in one place (App component) and pass it to child components. Don't use `import` keyword in non-root components. Here, you've correctly passed 'usersFromServer' to the form component within the 'App' component.